jax/_src/scipy/stats/betabinom.py

```diff
@@ -55,9 +55,11 @@ def logpmf(k: ArrayLike, n: ArrayLike, a: ArrayLike, b: ArrayLike,
   combiln = lax.neg(lax.add(lax.log1p(n), betaln(lax.add(lax.sub(n,y), one), lax.add(y,one))))
   beta_lns = lax.sub(betaln(lax.add(y,a), lax.add(lax.sub(n,y),b)), betaln(a,b))
   log_probs = lax.add(combiln, beta_lns)
-  y_cond = jnp.logical_or(lax.lt(y, lax.neg(loc)), lax.gt(y, lax.sub(n, loc)))
+  log_probs = jnp.where(jnp.logical_and(lax.eq(y, 0.), lax.eq(n, 0.)), 0., log_probs)
```

Comment:
You cannot use `lax.eq(n, 0.)` because `n` might not be the same dtype as `0.`.

your options are to use implicitly promoting operators (like `n == 0` rather than `lax.eq(n, 0.)`) or to explicitly construct constants of the right dtype (look for uses of `lax_internal._const`)

To hit these problematic issues locally, you can run a command like
```
JAX_ENABLE_X64=1 JAX_NUM_GENERATED_CASES=20 pytest -n auto tests/scipy_stats_test.py
```

Reply:
Modified accordingly and tested locally. Now passes all the tests related the changes in this PR. Squashed the commits into one.